pkg/controller/scheduledscaling/scheduled_scaling_test.go

```diff
@@ -297,3 +306,142 @@ func checkSchedules(t *testing.T, client zalandov1.ZalandoV1Interface, schedules
 	}
 	return nil
 }
+
+type mockScaler struct {
+	client kubernetes.Interface
+}
+
+func (s *mockScaler) Scale(ctx context.Context, hpa *autoscalingv2.HorizontalPodAutoscaler, replicas int32) error {
+	switch hpa.Spec.ScaleTargetRef.Kind {
+	case "Deployment":
+		deployment, err := s.client.AppsV1().Deployments(hpa.Namespace).Get(ctx, hpa.Spec.ScaleTargetRef.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		deployment.Spec.Replicas = &replicas
+		_, err = s.client.AppsV1().Deployments(hpa.Namespace).Update(ctx, deployment, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("unsupported kind %s", hpa.Spec.ScaleTargetRef.Kind)
+	}
+
+	return nil
+}
+
+func TestAdjustScaling(t *testing.T) {
+	for _, tc := range []struct {
+		msg             string
+		currentReplicas int32
+		desiredReplicas int32
+		targetValue     int64
+	}{
+		{
+			msg:             "current less than 10%% below desired",
+			currentReplicas: 95, // 5.3% increase to desired
+			desiredReplicas: 100,
+			targetValue:     10, // 1000/10 = 100
+		},
+		{
+			msg:             "current more than 10%% below desired, no adjustment",
+			currentReplicas: 90, // 11% increase to desired
+			desiredReplicas: 90,
```

Comment:
Shouldn't this be `100`?

Reply:
Maybe I can rename something to make it more clear. But the desired, from the perspective of the test case, is to make no adjustment because >10% we don't touch, but let the HPA deal with.

Reply:
Ah, got it! Within the scope of these tests the naming make sense, I was thinking too broad. I think we can leave it as is.